src/Containers/MinimalContainers/RecordArray.hpp

```diff
@@ -2,91 +2,54 @@
 // This file is distributed under the University of Illinois/NCSA Open Source License.
 // See LICENSE file in top directory for details.
 //
-// Copyright (c) 2020 QMCPACK developers.
+// Copyright (c) 2022 QMCPACK developers.
 //
 // File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//                    Ye Luo, yeluo@anl.gov, Argonne National Laboratory
 //
 // File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
 //////////////////////////////////////////////////////////////////////////////////////
 
 #ifndef QMCPLUSPLUS_RECORDARRAY_H
 #define QMCPLUSPLUS_RECORDARRAY_H
 
-#include <vector>
+#include "OhmmsPETE/OhmmsMatrix.h"
 
-// This is a data structure with two indices.  The data is a list of samples,
+// This is a data structure with two indices. The data is a list of samples,
 // and each sample has an entry for multiple parameters.
 // Think of a spreadsheet where the samples are the rows and the parameters are the columns.
-// This could be implemented with nested vectors (std::vector<std::vector<T>> records),
-// but that requires specifying the order of access everywhere it is used.  Any
-// changes to the storage or indexing order require changes everywhere.  This data
-// structure is an indirection between the specification of the access and the
-// underlying storage.
 //
-// In principle the underlying data could be switched from AoS (record-oriented)
-// to SoA (column-oriented), but the performance of accessing this structure is
-// unlikely to matter enough to warrant such functionality.
-//
-// The calls to setValue and getValue use the parameter index first, and the
-// sample entry index second.  It would nice to create different types for these
-// indices so the correct ordering could be enforced by the compiler.
-//
-
 namespace qmcplusplus
 {
 template<typename T>
-class RecordArray
+class RecordArray : public Matrix<T>
 {
 public:
-  void setValue(int param_idx, int entry_idx, T value)
-  {
-    storage_[param_idx * param_dim_ + entry_idx * nparam_] = value;
-  }
-
-  T getValue(int param_idx, int entry_idx) const
-  {
-    return storage_[param_idx * param_dim_ + entry_idx * entry_dim_];
-  }
+  using Base = Matrix<T>;
 
-  RecordArray() : nparam_(0), nentry_(0) {}
+  RecordArray() {}
 
   /// Constructor specifying the number of parameters and entries
 
-  RecordArray(int nparam, int nentry) : nparam_(nparam), nentry_(nentry)
```

Comment:
I have not swapped nparam and nentry order. I will push a commit to change it.
The documentation says entries are rows and parameters are columns. So the existing order is error-prone.

Reply:
changed.